unixfs/hamt/hamt.go

```diff
@@ -0,0 +1,393 @@
+package hamt
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"os"
+
+	proto "gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV/gogo-protobuf/proto"
+	"gx/ipfs/QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt/go-net/context"
+	"gx/ipfs/QmfJHywXQu98UeZtGJBQrPAR6AtmDjjbe3qjTo9piXHPnx/murmur3"
+
+	dag "github.com/ipfs/go-ipfs/merkledag"
+	format "github.com/ipfs/go-ipfs/unixfs"
+	upb "github.com/ipfs/go-ipfs/unixfs/pb"
+)
+
+const (
+	HashMurmur3 uint64 = iota
```

Comment:
it will likely use multihash if this proposal goes through: https://github.com/multiformats/multihash/issues/30